crates/flux-fixpoint/src/constraint.rs

```diff
@@ -360,31 +360,31 @@ impl<T: Types> fmt::Display for Expr<T> {
                 write!(f, "({func} {})", args.iter().map(FmtParens).format(" "),)
             }
             Expr::Neg(e) => {
-                write!(f, "-{}", FmtParens(e))
+                write!(f, "(- 0 {})", FmtParens(e))
```

Comment:
You are printing negative constants as `(- {val})`. Does that only work for literals? or can we change this to `write!("(- {})", FmtParens(e))`?

Reply:
Good catch, we don't need the `(- 0 {})` -- just `(- {})` will suffice!